### PR TITLE
Working on #2972 Template Conversion to Twig Format (index)

### PR DESCRIPTION
**Pull request by Shade-**
Working on #2972, converting index templates. My first conversion, hope everything's correct.

### CODE REVIEW

**Comment by euantorano** [comment]
The PR #2997 by @Matslom also targets `index.php`. I haven't looked at this code yet, but @Matslom was using raw HTML for the forum bit templates - if this PR converts those to twig too, we'll go with this PR rather than the other.

File: inc/views/base/index/index.twig [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+        <tbody style="{{ collapsed.boardstats_e }}" id="boardstats_e">
+            <tr>
+                <td class="tcat"><span class="smalltext"><strong>{{ lang.whos_online }}</strong> [<a href="online.php">{{ lang.complete_list }}</a>]</span></td>
+            </tr>
+            <tr>
+                <td class="trow1">
+                    <span class="smalltext">
+                        {{ lang.online_note }}<br />
+                        {% for user in users %}
+                        	{{ user.profilelink|raw }}{% if loop.last != true %}, {% endif %}
```
Comment (euantorano) [inline inc/views/base/index/index.twig]: This should use `{{ lang.comma }}` ratehr than a raw `,`.

File: inc/views/base/index/index.twig [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+                <td class="tcat">
+                    <span class="smalltext"><strong>{{ lang.todays_birthdays }}</strong></span>
+                </td>
+            </tr>
+            <tr>
+            	<td class="trow1">
+                    {% spaceless %}
+                    <span class="smalltext">
+                        {% for user in birthdays %}
+                        	{{ user.profilelink }}{% if user.age %} ({{ user.age }}) {% endif %}{% if loop.last != true %}, {% endif %}
```
Comment (euantorano) [inline inc/views/base/index/index.twig]: Doesn't this need `|raw` for the profile link?

Comment (euantorano) [reply inc/views/base/index/index.twig]: Also, again use `{{ lang.comma }}`.

File: inc/views/base/index/index.twig [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+                    <span class="smalltext"><strong>{{ lang.boardstats }}</strong></span>
+                </td>
+            </tr>
+            <tr>
+                <td class="trow1">
+                    <span class="smalltext">
+                        {{ lang.stats_posts_threads }}<br />
+                        {{ lang.stats_numusers }}<br />
+                        {{ lang.stats_newestuser|raw }}<br />
+                        {{ lang.stats_mostonline }}
```
Comment (euantorano) [inline inc/views/base/index/index.twig]: This should be `|raw` as it contains dates formatted using `my_date` which can contain HTML for relative dates.

File: index.php [hunk 4]
```diff
@@ -25,343 +25,261 @@
...
+    // Build a forum cache.
+    $query = $db->simple_select('forums', '*', 'active!=0', array('order_by' => 'pid, disporder'));
 
-	$forumsread = array();
-	if(isset($mybb->cookies['mybb']['forumread']))
-	{
-		$forumsread = my_unserialize($mybb->cookies['mybb']['forumread']);
-	}
+    $forumsread = array();
+    if (isset($mybb->cookies['mybb']['forumread']))
```
Comment (euantorano) [inline index.php]: This needs reformatting to have the `{` on the same line as the `if`.

**Comment by Shade-** [comment]
Ouch. Didn’t spot that @matslom already worked on it, I looked at the template list in #2972 and didn’t find index checked. Nevermind, I am going to update the list with the ones being worked on as it seems like there are some not checked yet. I am also going to fix those asap, thank you for reviewing!

**Comment by Shade-** [comment]
Oh, and this PR does not convert forums yet. Do you think they should be added here or as a separate PR?

**Comment by euantorano** [comment]
I'm not sure, I haven't looked at how big a job it is to convert them. Some of them are done (depth 2, which I needed for another page) but there's still a bunch to do.

**Comment by Shade-** [comment]
I'll handle them in a separate PR as it looks neater.

**Comment by euantorano** [comment]
Ok, fair enough. Question is whether to merge this one or the other now, as both achieve the same thing 😉

**Pull request merged**